src/platform/datapath_raw_xdp.c

```diff
@@ -690,6 +758,81 @@ CxPlatDpRawInitialize(
         }
     }
 
+Error:
+    if (QUIC_FAILED(Status)) {
+        CxPlatDpRawInterfaceUninitialize(Interface);
+    }
+
+    return Status;
+}
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+CxPlatDpRawInitialize(
+    _Inout_ CXPLAT_DATAPATH* Datapath,
+    _In_ uint32_t ClientRecvContextLength
+    )
+{
+    XDP_DATAPATH* Xdp = (XDP_DATAPATH*)Datapath;
+    CXPLAT_THREAD_CONFIG Config = {
+        0, 0, "XdpDatapathWorker", CxPlatXdpWorkerThread, Xdp
+    };
+    QUIC_STATUS Status;
+    MIB_IF_TABLE2* IfTable;
+    MIB_IF_ROW2* IfRow;
+
+    CxPlatXdpReadConfig(Xdp);
+    Datapath->Cpu = Xdp->DatapathCpuNumber;
+    CxPlatDpRawGenerateCpuTable(Datapath);
+    CxPlatListInitializeHead(&Xdp->Interfaces);
+
+    Status = GetIfTable2(&IfTable);
+    if (QUIC_FAILED(Status)) {
+        QuicTraceEvent(
+            LibraryErrorStatus,
+            "[ lib] ERROR, %u, %s.",
+            Status,
+            "GetIfTable2");
+        goto Error;
+    }
+
+    for (uint32_t i = 0; i < IfTable->NumEntries; i++) {
+        IfRow = (MIB_IF_ROW2 *)&IfTable->Table[i];
+        if (!IfRow->InterfaceAndOperStatusFlags.FilterInterface &&
+            !IfRow->InterfaceAndOperStatusFlags.NotMediaConnected &&
+            !IfRow->InterfaceAndOperStatusFlags.Paused &&
+            IfRow->OperStatus == IfOperStatusUp &&
+            IfRow->MediaType == NdisMedium802_3) {
+            XDP_INTERFACE* Interface = CxPlatAlloc(sizeof(XDP_INTERFACE), IF_TAG);
+            if (Interface == NULL) {
+                QuicTraceEvent(
+                    AllocFailure,
+                    "Allocation of '%s' failed. (%llu bytes)",
+                    "XDP interface",
+                    sizeof(*Interface));
+                continue;
+            }
+
+            CxPlatZeroMemory(Interface, sizeof(*Interface));
+
+            Status =
+                CxPlatDpRawInterfaceInitialize(
+                    Xdp, Interface, IfRow->InterfaceIndex, ClientRecvContextLength);
+            if (QUIC_FAILED(Status)) {
+                QuicTraceEvent(
+                    LibraryErrorStatus,
+                    "[ lib] ERROR, %u, %s.",
+                    Status,
+                    "CxPlatDpRawInterfaceInitialize");
+                CxPlatFree(Interface, IF_TAG);
+                continue;
+            }
+
+            printf("Bound XDP to interface %u\n", IfRow->InterfaceIndex);
```

Comment:
For now, wrap any printfs you think we need to keep in `#if DEBUG`, otherwise remove them please.